src/pydantic_spark/base.py

```diff
@@ -62,12 +58,15 @@ def get_type_of_definition(ref: str, schema: dict):
         def get_type(value: dict) -> Tuple[str, dict]:
             """Returns a type of single field"""
             t = value.get("type")
+            ao = value.get("anyOf")
             f = value.get("format")
             r = value.get("$ref")
             a = value.get("additionalProperties")
-            e = value.get("json_schema_extra", {})
-            ft = e.get("coerce_type")
+            ft = value.get("coerce_type")
             metadata = {}
+            if ao is not None:
+                t = ao[0].get("type")
```

Comment:
This does mean it only takes the first value? If anyOf is used I do assume multiple types for a single field are there. Maybe better to even not allow for this now. Might give side effects without people will know about this.

Reply:
AnyOf is also used for Optional columns. Adjusted logic to handle this case.

Reply:
looks good now